test/test_common/utility.h

```diff
@@ -371,6 +374,27 @@ class TestAllocator : public RawStatDataAllocator {
   std::unordered_map<std::string, CSmartPtr<RawStatData, freeAdapter>> stats_;
 };
 
+class MockedTestAllocator : public RawStatDataAllocator {
```

Comment:
Note: this mocked allocator was previously repeated 3x in the codebase, all as part of multiple inheritance with a test class. What motivated the refactor in this PR, though, was the need to pass a StatsOptions& to the constructor, which is annoying to do for a test-class.

If desired I could split that out as a separate PR.